common/djangoapps/student/auth.py

```diff
@@ -90,6 +92,9 @@ def get_user_permissions(user, course_key, org=None):
         return all_perms
     if course_key and user_has_role(user, CourseInstructorRole(course_key)):
         return all_perms
+    # User has all permission for libraries and courses if user belongs to content creator role.
+    if OrgContentCreatorRole(org=org).has_user(user):
+        return all_perms
```

Comment:
Because when CourseContentGroup is granted user is given a `Staff` and `Instructor` role. But for the specific organization we give the user the `OrgContentCreatorRole` so we had to introduce the new role here. Hope I am able to clarify this.

Reply:
> Why do we need to do this for OrgContentCreatorRole if we don't do it for CourseCreatorRole?

> Because when CourseContentGroup is granted user is given a Staff and Instructor role. But for the specific organization we give the user the OrgContentCreatorRole so we had to introduce the new role here. Hope I am able to clarify this.

@farhaanbukhsh Sorry, I don't fully understand. Are you saying that when `CourseCreatorRole` is granted to someone, they are given the `CourseStaff` and `CourseInstructor` access roles? How would that work when course-creator is a global role, and staff/instructor are course-specific roles?

Reply:
@kdmccormick `CourseStaff` and `CourseInstructor` access roles are assigned to the person who creates the role.

The check here is necessary because `OrgContentCreatorRole` deals with the creation of course so we need to raise error when user not belonging to the org. is creating a course.

While CourseCreatorRole is required to access `course creation` page and not check anything to do with course access. Hence we require the check for organization.

Hope I am able to clarify this.

Reply:
@farhaanbukhsh This is still unclear to me. I cannot find any code paths that provide `org` as an argument to `get_user_permission`. Could you provide to me a specific scenario in which having this check here is necessary?

Reply:
`org` is not passed as a parameter to `get_user_permission` but deduced from course key [here](https://github.com/edx/edx-platform/blob/master/common/djangoapps/student/auth.py#L80), hence there is no need for explicitly passing org.



> Could you provide to me a specific scenario in which having this check here is necessary?

This is a very crucial check where write access for a user is checked for creating courses. You can have a look at this [test case](https://github.com/edx/edx-platform/pull/26616/files#diff-87512b8ef163801a02115105c1fc230d197ca6a00e4185cef51f2181212dc399R301) where write access is based on the role given to the user.

This check is indirectly used in a lot of places but the code path that touches this function in this PR starts [here](https://github.com/edx/edx-platform/blob/master/cms/djangoapps/contentstore/views/library.py#L283) and  [here](https://github.com/edx/edx-platform/blob/master/cms/djangoapps/contentstore/views/course.py#L283)

This is a non-exhaustive list.

Hope it's clearer now.

Reply:
@farhaanbukhsh 
> This check is indirectly used in a lot of places but the code path that touches this function in this PR starts here and here

I tried removing this check, and found that the course creator still has instructor+staff access to the course. This is because we explicitly [grant instructor access](https://github.com/edx/edx-platform/blob/d5bacad86f9e37a04d58b26335fcd1967ffd99b8/cms/djangoapps/contentstore/views/course.py#L964) to a user when they create a course, rerun a course, or create a library.

Reply:
Tracing back the change I feel you are right, this check is not necessary now. It's possibly a left over from some approach that didn't work. I am sorry. :(

Reply:
No sweat! Glad we worked through it.